ecl/baremetal/v2/server.py

```diff
@@ -155,9 +155,12 @@ def start(self, session, server_id, boot_mode):
         self._translate_response(resp, has_body=False)
         return self
 
-    def stop(self, session, server_id):
+    def stop(self, session, server_id, type):
         uri = self.base_path % server_id
-        body = {"os-stop": None}
+        if type:
```

Comment:
I think this is better when checking if a value is None or not.
```suggestion
        if type is None:
            body = {"os-stop": None}
        else:
            body = {"os-stop": {"type": type}}
```